tensorflow_datasets/core/utils/py_utils.py

```diff
@@ -363,6 +363,14 @@ def reraise(
     else:
       exception = RuntimeError(f'{type(e).__name__}: {msg}')
     raise exception from e
+  #Raise a custom error message for datasets using Array2D from huggingface
+  elif isinstance(e, AttributeError) and 'Array2D' in e.args[0]:
```

Comment:
This is not the right place to add this. When coding it's important to keep abstraction layers.

Array2D is specific to hugging face, so should be added in hugging face module in `community/`.
`py_utils` module is a very generic module which should not have any assumption the error message raised